framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -275,6 +275,34 @@
     protected AnnotationClassLoader loader;
 
     /**
+     * Should results be cached?
+     * This means that ATM.deepCopy() will be called.
+     * ATM.deepCopy() use to (and perhaps still does) side effect the ATM being copied.
+     * So setting this to false is not equivalent to setting shouldReadCache to false.
+     * */
+    public boolean shouldCache;
```

Comment:
Why are fields shouldCache and shouldReadCache public?
Can you document why they are not final?